tests/test_validation_framework/test_validation.py

```diff
@@ -426,6 +426,146 @@ def test_ascat_ismn_validation_metadata(ascat_reader):
+    )
+    ismn_reader = ISMN_Interface(ismn_data_folder)
+
+    # Create the variable ***save_path*** which is a string representing the
+    # path where the results will be saved. **DO NOT CHANGE** the name
+    # ***save_path*** because it will be searched during the parallel
+    # processing!
+
+    save_path = tempfile.mkdtemp()
+
+    # Create the validation object.
+
+    datasets = {
+        "ASCAT": {
+            "class": ascat_reader,
+            "columns": ["sm"],
+            "kwargs": {
+                "mask_frozen_prob": 80,
+                "mask_snow_prob": 80,
+                "mask_ssf": True,
+            }
+        },
+        "ISMN": {
+            "class": ismn_reader,
+            "columns": ["soil moisture"],
+        },
+    }
+
+    read_ts_names = {"ASCAT": "read", "ISMN": "read_ts"}
+    period = [datetime(2007, 1, 1), datetime(2014, 12, 31)]
+
+    datasets = DataManager(
+        datasets,
+        "ASCAT",
+        period,
+        read_ts_names=read_ts_names,
+        upscale_parms={
+            "upscaling_method": "average",
+            "temporal_stability": True,
+            "upscaling_lut": upscaling_lut,
+        },
+    )
+    process = Validation(
+        datasets,
+        "ASCAT",
+        temporal_ref="ISMN",
+        scaling="lin_cdf_match",
+        scaling_ref="ISMN",
+        metrics_calculators={
+            (2, 2): metrics_calculators.BasicMetrics(
+                other_name="k1"
+            ).calc_metrics
+        },
+        period=period,
+    )
+
+    for job in jobs:
+        results = process.calc(*job)
+        netcdf_results_manager(results, save_path)
+
+    results_fname = os.path.join(
+        save_path, "ASCAT.sm_with_ISMN.soil moisture.nc"
+    )
+
+    with nc.Dataset(results_fname, mode="r") as results:
+        calc_vars = results.variables.keys()
+        n_obs = results.variables["n_obs"][:].tolist()
+        rho = results.variables["rho"][:]
+        rmsd = results.variables["RMSD"][:]
+        r = results.variables["R"][:]
+
+    vars_should = [
+        u"n_obs",
+        u"tau",
+        u"gpi",
+        u"RMSD",
+        u"lon",
+        u"p_tau",
+        u"BIAS",
+        u"p_rho",
+        u"rho",
+        u"lat",
+        u"R",
+        u"p_R",
+        u"time",
+        u"idx",
+        u"_row_size",
+    ]
+    n_obs_should = [764, 2392, 904]
+
+    rho_should = np.array(
+        [-0.012487,  0.255156,  0.635517],
+        dtype=np.float32,
+    )
+    rmsd_should = np.array(
+        [0.056428, 0.056508, 0.116294],
+        dtype=np.float32,
+    )
+    r_should = np.array(
+        [-0.012335,  0.257671,  0.657239],
+        dtype=np.float32,
+    )
+
+    assert sorted(vars_should) == sorted(calc_vars)
+    assert n_obs == n_obs_should
+    nptest.assert_allclose(sorted(rho), sorted(rho_should), rtol=1e-4)
+    nptest.assert_allclose(sorted(rmsd), sorted(rmsd_should), rtol=1e-4)
+    nptest.assert_allclose(sorted(r), sorted(r_should), rtol=1e-4)
```

Comment:
This is the same as in other tests, could you make a function out of this instead?

Reply:
I have created this function, but the values need to be specified test-by-test regardless :/ ; still, a bit cleaner in the code